packages/compass-workspace/src/components/tab.tsx

```diff
@@ -0,0 +1,314 @@
+  backgroundColor: uiColors.focus,
+  transition: 'height 150ms ease-out',
+});
+
+const hiddenStyles = css({
+  visibility: 'hidden',
+});
+
+const tabIconStyles = css({
+  width: spacing[1] + spacing[2],
+  height: 'auto',
+  flexShrink: 0,
+  paddingBottom: spacing[3] - 1,
+});
+
+const tabIconSelectedLightThemeStyles = css({
+  color: uiColors.green.dark2,
+});
+
+const tabIconSelectedDarkThemeStyles = css({
+  color: uiColors.green.light2,
+});
+
+const tabIconFocusedStyles = css({
+  color: uiColors.focus,
+});
+
+const tabTitleContainerStyles = css({
+  marginLeft: spacing[2],
+  marginRight: spacing[1],
+  display: 'inline-grid',
+  gridTemplateColumns: '1fr',
+});
+
+const tabTitleStyles = css({
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  overflow: 'hidden',
+  fontWeight: 'bold',
+  fontSize: compassFontSizes.smallFontSize,
+});
+
+const tabTitleDarkThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabTitleLightThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabTitleSelectedDarkThemeStyles = css({
+  color: uiColors.green.light2,
+});
+
+const tabTitleSelectedLightThemeStyles = css({
+  color: uiColors.green.dark2,
+});
+
+const tabTitleFocusedStyles = css({
+  color: uiColors.focus,
+});
+
+const tabSubtitleStyles = css({
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  overflow: 'hidden',
+  fontSize: compassFontSizes.smallFontSize,
+  lineHeight: `${spacing[3]}px`,
+});
+
+const tabSubtitleLightThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabSubtitleDarkThemeStyles = css({
+  color: uiColors.gray.light1,
+});
+
+const tabSubtitleSelectedLightThemeStyles = css({
+  color: uiColors.gray.dark1,
+});
+
+const tabSubtitleSelectedDarkThemeStyles = css({
+  color: uiColors.gray.light2,
+});
+
+type TabProps = {
+  activeSubTabName: string;
+  darkMode?: boolean;
+  isFocused: boolean;
+  isSelected: boolean;
+  onSelect: () => void;
+  onClose: () => void;
+  tabId: string;
+  namespace: string;
+  type: TabType;
+  isTabListFocused: boolean;
+};
+
+function UnthemedTab({
+  darkMode,
+  activeSubTabName,
+  isFocused,
+  isSelected,
+  isTabListFocused,
+  onSelect,
+  onClose,
+  tabId,
+  type,
+  namespace,
+}: TabProps) {
+  const defaultActionProps = useDefaultAction(onSelect);
+
+  const [hoverProps, isHovered] = useHoverState();
+
+  const tabProps = mergeProps<HTMLDivElement>(hoverProps, defaultActionProps);
+
+  const tabIcon = useMemo(() => {
+    return type === 'timeseries'
+      ? 'TimeSeries'
+      : type === 'view'
+      ? 'Visibility'
+      : 'Folder';
+  }, [type]);
+
+  return (
+    <div
+      className={cx(
+        tabStyles,
+        darkMode ? tabDarkThemeStyles : tabLightThemeStyles,
+        {
+          [selectedTabStyles]: isSelected,
+          [focusedTabStyles]: isFocused,
+        }
+      )}
+      aria-selected={isSelected}
+      role="tab"
+      // The tab navigation is handled by the lab list.
+      tabIndex={-1}
```

Comment:
Because tabIndex never changes on these elements when using keyboard to navigate them with VO enabled you never get informed that the tab is changed

Reply:
Updated - added a `useRovingTabIndex` for focusing the tabs, and now only the selected/active tab will have `tabIndex={0}`. This `useRovingTabIndex` could live in `compass-components`, but since this is the only use place of it I haven't moved it there yet. Would you prefer I move it over now? Otherwise we can just keep an eye on it and move it over if we end up needing it elsewhere.

Reply:
<img width="540" alt="Screen Shot 2022-03-16 at 10 45 40 PM" src="https://user-images.githubusercontent.com/1791149/158726331-aa4307d0-bc18-46d1-a6cf-817c39b2c876.png">

Reply:
I think if it's something that only used in one place it would make sense not to move it. Kind of a tangent, but still related: I do feel like now with the updated, more generic tabs interface that this component has this whole thing can be part of compass-components and not a separate package, would that make sense to you?